menas/ui/service/RunService.js

```diff
@@ -0,0 +1,215 @@
+      })
+  };
+
+  this.getLatestRunForLatestVersion = function (oControl, oTable, datasetName) {
+    Functions.ajax("api/runs/" + encodeURI(datasetName) + "/latest", "GET", {},
+      oData => {
+        this.setCurrentRun(oControl, oTable, oData);
+      },
+      () => {
+        sap.m.MessageBox
+          .error("Failed to get the list of runs. Please wait a moment and try reloading the application")
+      })
+  };
+
+  this.setCurrentRun = function (oControl, oTable, oRun) {
+    let checkpoints = oRun.controlMeasure.checkpoints;
+
+    this._preprocessRun(oRun, checkpoints);
+    this._setUpCheckpointsTable(checkpoints, oTable);
+
+    oControl.setModel(new sap.ui.model.json.JSONModel(oRun), "run");
+    oControl.setModel(new sap.ui.model.json.JSONModel(oRun.controlMeasure.metadata), "metadata");
+  };
+
+  this._bindRunSummaries = function(oRunSummaries, oControl) {
+    oRunSummaries.forEach(run => {
+      run.status = this._normalizeStatus(run.status)
+    });
+    oControl.setModel(new sap.ui.model.json.JSONModel(oRunSummaries), "runs");
+  };
+
+  this._preprocessRun = function (oRun, aCheckpoints) {
+    let info = oRun.controlMeasure.metadata.additionalInfo;
+    oRun.controlMeasure.metadata.additionalInfo = this._mapAdditionalInfo(info);
+
+    oRun.status = this._normalizeStatus(oRun.runStatus.status.value);
+
+    oRun.stdTime = this._getTimeSummary(aCheckpoints, "Standardization Finish", "Standardization Finish");
+    oRun.cfmTime = this._getTimeSummary(aCheckpoints, "Conformance - Start", "Conformance - End");
+  };
+
+  this._mapAdditionalInfo = function (info) {
+    return Object.keys(info).map(key => {
+      return {"infoKey": key, "infoValue": info[key]}
+    });
+  };
+
+  this._normalizeStatus = function(sStatus) {
+    switch(sStatus) {
+      case "failed" :
+        return "Failed";
+      case "running" :
+        return "Running";
+      case "stageSucceeded" :
+        return "Stage Succeeded";
+      case "allSucceeded" :
+        return "All Succeeded";
+    }
+  };
+
+  this._getTimeSummary = function (aCheckpoints, sStartCheckpoint, sEndCheckpoint) {
+    let startDateTime = this._findStartTime(aCheckpoints, sStartCheckpoint);
+    let endDateTime = this._findEndTime(aCheckpoints, sEndCheckpoint);
+    let duration = this._getDurationAsString(startDateTime, endDateTime);
+
+    return {
+      startDateTime: startDateTime,
+      endDateTime: endDateTime,
+      duration: duration
+    }
+  };
+
+  this._setUpCheckpointsTable = function (checkpoints, oTable) {
+    checkpoints.forEach(checkpoint => {
+      checkpoint.duration = this._getDurationAsString(checkpoint.processStartTime, checkpoint.processEndTime);
+    });
+
+    let controlNames = checkpoints.flatMap(checkpoint =>
+      checkpoint.controls.map(control => {
+        checkpoint[control.controlName] = control.controlValue;
+        return control.controlName;
+      })
+    );
+
+    oTable.removeAllColumns();
```

Comment:
Could this be done in the view?

Reply:
I tried... I couldn't come up with a way to do it since the columns need to be dynamically generated, which I was only able to figure to do programatically.